channel/consensus_channel/consensus_channel.go

```diff
@@ -457,6 +504,58 @@ func (vars *Vars) Add(p Add) error {
 	return nil
 }
 
+// Remove mutates Vars by
+// - increasing the turn number by 1
+// - removing the guarantee for the Target channel
+// - adjusting balances accordingly based on LeftAmount and RightAmount
+//
+// An error is returned if:
+// - the turn number is not incremented
+// - a guarantee is not found for the target
+// - the amounts are too large for the guarantee amount
+//
+// If an error is returned, the original vars is not mutated
+func (vars *Vars) Remove(p Remove) error {
+	// CHECKS
+
+	if p.turnNum != vars.TurnNum+1 {
+		return ErrIncorrectTurnNum
+	}
+	o := vars.Outcome
+
+	guarantee, found := o.guarantees[p.Target]
+	if !found {
+		return ErrGuaranteeNotFound
+	}
+
+	totalRemoved := big.NewInt(0).Add(p.LeftAmount, p.RightAmount)
+	if totalRemoved.Cmp(guarantee.amount) != 0 {
+		return ErrInsufficientFunds
+	}
+
+	// EFFECTS
+
+	// Increase the turn number
+	vars.TurnNum += 1
+
+	// Adjust balances
+	o.left.amount.Add(o.left.amount, p.LeftAmount)
+	o.right.amount.Add(o.right.amount, p.RightAmount)
+
+	// Remove the guarantee
+	delete(o.guarantees, p.Target)
+
+	return nil
+}
+
+// Remove is a proposal to remover a guarantee for the given virtual channel
+type Remove struct {
+	turnNum     uint64
+	Target      types.Destination
+	LeftAmount  *big.Int
+	RightAmount *big.Int
```

Comment:
What do `LeftAmount`, `RightAmount` refer to?

I think it's the balances that should be settled with -- is there a better name or some Godoc that could clarify?

Reply:
These are the amounts that should be credited to the `left` and `right` participants of a two-party ledger channel when removing the guarantee.

I agree that `LeftAmount` and `RightAmount` aren't super clear; do you have any name suggestions?

Reply:
Maybe `LeftRefund`/`RightRefund`? Otherwise, putting this in a godoc could do:
> These are the amounts that should be credited to the left and right participants of a two-party ledger channel when removing the guarantee.

Reply:
I have added some godoc in [228c391](https://github.com/statechannels/go-nitro/pull/483/commits/228c3913381150cc4a35b7579ffade24b50cb976)